run/events_pubsub/main.go

```diff
@@ -12,38 +12,49 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// [START run_events_pubsub_server]
+// [START run_events_pubsub_handler]
 
-// Sample run-pubsub-events is a Cloud Run service which handles Pub/Sub messages.
+// Sample run-events-pubsub is a Cloud Run service which handles Pub/Sub messages.
 package main
 
 import (
-	"context"
+	b64 "encoding/base64"
+	"encoding/json"
 	"fmt"
 	"log"
 	"net/http"
 	"os"
-
-	cloudevents "github.com/cloudevents/sdk-go/v2"
 )
 
-var (
-	handler = http.DefaultServeMux
-)
+// PubSubMessage is the payload of a Pub/Sub event.
+type PubSubMessage struct {
+	Message struct {
+		Data string `json:"data,omitempty"`
+		ID   string `json:"id"`
+	} `json:"message"`
+	Subscription string `json:"subscription"`
+}
 
-func main() {
-	ctx := context.Background()
-	// Create a new HTTP client for CloudEvents
-	p, err := cloudevents.NewHTTP()
-	if err != nil {
-		log.Fatal(err)
+// HelloEventsPubSub receives and processes a Pub/Sub push message.
+func HelloEventsPubSub(w http.ResponseWriter, r *http.Request) {
+	var e PubSubMessage
+	if err := json.NewDecoder(r.Body).Decode(&e); err != nil {
+		http.Error(w, "Bad Request", http.StatusBadRequest)
```

Comment:
Might want to log this error so it shows up somewhere.

Reply:
Sure, added log.